cmd/kubeadm/app/phases/certs/certs.go

```diff
@@ -548,5 +548,10 @@ func getAltNames(cfg *kubeadmapi.MasterConfiguration) (*certutil.AltNames, error
 		}
 	}
 
+	// add api server dns advertise address
+	if len(cfg.API.ControlPlaneEndpoint) > 0 {
```

Comment:
Shouldn't we also validate that `ControlPlaneEndpoint` is a proper DNS subdomain, i.e., `validation.IsDNS1123Subdomain(ControlPlaneEndpoint)`?

Reply:
I added this and also covered with tests.